hack/generate/samples/pkg/context.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/test/utils"
+)
+
+// SampleContext represents the Context used to generate the samples
+type SampleContext struct {
+	utils.TestContext
+}
+
+// NewSampleContext returns a SampleContext containing a new kubebuilder TestContext.
+func NewSampleContext(binary string, path string, env ...string) (s SampleContext, err error) {
+	s.TestContext, err = utils.NewTestContext(binary, env...)
+	// If the path was informed then this should be the dir used
+	if strings.TrimSpace(path) != "" {
+		path, err = filepath.Abs(path)
+		if err != nil {
+			return s, err
+		}
+		s.CmdContext.Dir = path
+		s.ProjectName = strings.ToLower(filepath.Base(s.Dir))
+		s.ImageName = fmt.Sprintf("quay.io/example/%s:v0.0.1", s.ProjectName)
+		s.BundleImageName = fmt.Sprintf("quay.io/example/%s-bundle:v0.0.1", s.ProjectName)
+	}
+
+	return s, err
+}
+
+// NewSampleContextWith returns a SampleContext containing the kubebuilder TestContext informed
+func NewSampleContextWith(tc *utils.TestContext) (s SampleContext, err error) {
```

Comment:
I would rename this `WithTestContext`

Reply:
Actually what is using that method? I don't see any other caller of this code.

Reply:
It will be used to re-use the samples code in the e2e tests. See: 


**To call in the e2e test**
```go
By("running samples steps")
ctx, err := samplespkg.NewSampleContextWith(&tc)
Expect(err).Should(Succeed())
sample := sampleshelm.NewMemcachedHelm(&ctx)
sample.Run()
```
ref: https://github.com/operator-framework/enhancements/pull/47/files